src/Language/PureScript/TypeChecker/Entailment.hs

```diff
@@ -206,9 +206,14 @@ entails SolverOptions{..} constraint context hints =
             -- We need information about functional dependencies, so we have to look up the class
             -- name in the environment:
             classesInScope <- lift . lift $ gets (typeClasses . checkEnv)
-            TypeClassData{ typeClassDependencies } <- case M.lookup className' classesInScope of
-              Nothing -> throwError . errorMessage $ UnknownClass className'
-              Just tcd -> pure tcd
+
+            TypeClassData
+              { typeClassDependencies
+              , typeClassIsEmpty
+              } <- case M.lookup className' classesInScope of
+                Nothing -> throwError . errorMessage $ UnknownClass className'
```

Comment:
Is this a sign for a programming error in the compiler, or an actual user facing error?

In the first case I'd prefer an internalError, and in the second case this really ought to have a source span (unless this is wrapped somewhere and gets a source span)

Reply:
It's the first case, I believe.  Will change to `internalError`, thanks!

Reply:
Actually, turns out, it's not the first case. :sweat_smile:

Reply:
The errors appear to be propagated through `typesOf` and into the type checker; where source spans get added and the errors re-thrown.